spec/Section 6 -- Execution.md

```diff
@@ -42,44 +73,106 @@ of variable's declared type. If a query error is encountered during
 input coercion of variable values, then the operation fails without
 execution.
 
+If any variable defined as non-null is not provided, or is provided the value
+{null}, then the operation fails without execution.
 
-## Evaluating operations
+CoerceVariableValues(schema, operation, variableValues)
+
+## Executing Operations
 
 The type system, as described in the “Type System” part of the spec, must
 provide a “Query Root” and a “Mutation Root” object.
 
+If the operation is a query, the result of the operation is the result of
+executing the query’s top level selection set on the “Query Root” object.
+
+An initial value can be optionally provided when executing a query.
+
+ExecuteQuery(query, schema, variableValues, initialValue):
+
+  * Let {queryType} be the root Query type in {schema}.
+  * Assert: {queryType} is an Object type.
+  * Let {selectionSet} be the top level Selection Set in {query}.
+  * Let {data} be the result of running
+    {ExecuteSelectionSet(selectionSet, queryType, initialValue, variableValues)}
+    *normally* (allowing parallelization).
+  * Let {errors} be any *field errors* produced while executing the
+    selection set.
+  * Return an unordered map containing {data} and {errors}.
+
 If the operation is a mutation, the result of the operation is the result of
-evaluating the mutation’s top level selection set on the “Mutation Root”
-object. This selection set should be evaluated serially.
+executing the mutation’s top level selection set on the “Mutation Root”
+object. This selection set should be executed serially.
 
-If the operation is a query, the result of the operation is the result of
-evaluating the query’s top level selection set on the “Query Root” object.
+It is expected that the top level fields in a mutation operation perform
+side-effects on the underlying data system. Serial execution of the provided
+mutations ensures against race conditions during these side-effects.
+
+ExecuteMutation(mutation, schema, variableValues, initialValue):
+
+  * Let {variableValues} be the set of variable values to be used by any
+    field argument value coercion.
```

Comment:
This step seems redundant, doesn't say anything useful (and we don't have it in the "query" algorithm above).